src/plot_api/plot_api.js

```diff
@@ -266,10 +266,12 @@ Plotly.plot = function(gd, data, layout, config) {
         // clean up old scenes that no longer have associated data
         // will this be a performance hit?
 
-        // ... until subplot of different type play better together
-        if(gd._fullLayout._hasGL3D) plotGl3d(gd);
-        if(gd._fullLayout._hasGeo) plotGeo(gd);
-        if(gd._fullLayout._hasGL2D) plotGl2d(gd);
+        var registry = plots.subplotsRegistry;
+
+        // TODO incorporate cartesian and polar plots into this paradigm
+        if(fullLayout._hasGL3D && registry.gl3d) registry.gl3d.plot(gd);
+        if(fullLayout._hasGeo && registry.geo) registry.geo.plot(gd);
+        if(fullLayout._hasGL2D && registry.gl2d) registry.gl2d.plot(gd);
```

Comment:
I can't imagine a scenario where that would be a thing. Probably just checking for `_hasGL3d` is sufficient. It would probably be better for debugging as well to see `registry.gl3d.plot is not a function` instead of silent failure too.